torch/csrc/jit/backends/backend.h

```diff
@@ -0,0 +1,155 @@
+  // method_compile_spec).
+  virtual c10::impl::GenericDict compile(
+      c10::IValue processed,
+      c10::impl::GenericDict method_compile_spec) = 0;
+
+  // Execute the method specified by the last argument on \p stack. The rest of
+  // the stack contents are inputs.
+  virtual void execute(Stack& stack) = 0;
+};
+
+c10::FunctionSchema getPreprocessSchema() {
+  c10::Argument self("self", c10::AnyType::get());
+  c10::Argument mod("mod", c10::AnyType::get());
+  c10::Argument method_compile_spec(
+      "method_compile_spec",
+      c10::DictType::create(c10::StringType::get(), c10::AnyType::get()));
+
+  c10::FunctionSchema preprocessor_schema(
+      "preprocess",
+      /*overload_name=*/"",
+      /*arguments=*/{self, mod, method_compile_spec},
+      /*returns=*/{mod});
+  return preprocessor_schema;
+}
+
+template <typename TBackendInterface>
+std::function<void(Stack&)> getPreprocessFunc() {
+  return [](Stack& stack) {
+    auto method_compile_spec = pop(stack).toGenericDict();
+    auto mod = pop(stack);
+    auto self = pop(stack).toCustomClass<TBackendInterface>();
+    auto ret = self->preprocess(mod, method_compile_spec);
+    push(stack, ret);
+  };
+}
+
+c10::FunctionSchema getCompileSchema() {
+  c10::Argument self("self", c10::AnyType::get());
+  c10::Argument mod("processed", c10::AnyType::get());
+  auto any_dict_ty =
+      c10::DictType::create(c10::StringType::get(), c10::AnyType::get());
+  c10::Argument method_compile_spec("method_compile_spec", any_dict_ty);
+  c10::Argument handles("handles", any_dict_ty);
+
+  c10::FunctionSchema compile_schema(
+      "compile",
+      /*overload_name=*/"",
+      /*arguments=*/{self, mod, method_compile_spec},
+      /*returns=*/{handles});
+  return compile_schema;
+}
+
+template <typename TBackendInterface>
+std::function<void(Stack&)> getCompileFunc() {
+  return [](Stack& stack) {
+    auto method_compile_spec = pop(stack).toGenericDict();
+    auto processed = pop(stack);
+    auto self = pop(stack).toCustomClass<TBackendInterface>();
+    auto ret = self->compile(processed, method_compile_spec);
+    push(stack, ret);
+  };
+}
+
+c10::FunctionSchema getExecuteSchema() {
+  c10::Argument mod("processed", c10::AnyType::get());
+  // TODO: This is actually not accurate because execute is varret, but you can
+  // only use a FunctionSchema with one return type to create a
+  // jit::BuiltinOpFunction (which is what) torch::class_ does under the hood.
+  return c10::FunctionSchema(
+      "execute",
+      /*overload_name=*/"",
+      /*arguments=*/{},
+      /*returns=*/{mod},
+      true,
+      true);
+}
+
+template <typename TBackendInterface>
+std::function<void(Stack&)> getExecuteFunc() {
+  return [](Stack& stack) {
+    // TODO: Stack does not have the number of arguments on it (probably because
+    // schema is not truly varret) so the best we can do is to assume that
+    // everything on the stack is an argument to execute.
+    auto num_args = stack.size();
```

Comment:
The interpreter stack might contain other stuff (it's a stack machine so it uses the stack to hold intermediates and such), so `stack.size()` does not work; you are only supposed to pop off the arguments you know that you need. Let me look into why we're not getting the number of arguments.